assets/differentialabundance_report.Rmd

```diff
@@ -382,19 +389,43 @@ differential_results <- lapply(differential_files, function(diff_file){
         qval_column = params$differential_qval_column
     )
 
-    # If fold changes are not logged already, log them (we assume they're logged
-    # later on)
-
-    if (! params$differential_foldchanges_logged){
+    # If fold changes are not logged already, log them
+    if (!params$differential_foldchanges_logged) {
         diff[[params$differential_fc_column]] <- log2(diff[[params$differential_fc_column]])
     }
 
     # Annotate differential tables if possible
-    if (! is.null(params$features)){
-        diff <- merge(features, diff, by.x = params$features_id_col, by.y = params$differential_feature_id_column)
+    if (!is.null(params$features)) {
+
+        # Merge tables
+        diff_features <- merge(features, diff, by.x = params$features_id_col, by.y = params$differential_feature_id_column)
```

Comment:
The warnings are great. But I am wondering what would be the consequence of
```suggestion
        diff_features <- merge(features, diff, by.x = params$features_id_col, by.y = params$differential_feature_id_column, all.x = FALSE, all.y = TRUE)
```
would that lead to a complete plot (all features present) but with NA for annotations? If so, that might be a nice solution as well.

Reply:
@d4straub As I can see it, the volcano plots won''t be affected, but I think this should be reported anyway (number of missing rows or number of features with NAs). We might take care of the new plots recently added below too (still in dev). I'm not sure if they can handle tables with NAs or without these columns at all @atrigila